File: Chapter-2/2/count.js
// node count.js

const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.question('What is the input string? ', (s) => {
  const length = s.length;
  if (length == 0) {
    console.log('Empty string!');
  } else {
    console.log('%s has %d characters', s, length);
  }
  rl.close();
});

File: Chapter-2/2/count.go
// go run count.go

package main

import (
  "fmt"
  "bufio"
  "os"
  "strings"
)

func main() {
  fmt.Print("What is the input string? ")
  reader := bufio.NewReader(os.Stdin)
  s, _ := reader.ReadString('\n')
  trimmed := strings.TrimSpace(s)
  count := len(trimmed)

  if count == 0 {
    fmt.Printf("Empty string!\n")
  } else {
    fmt.Printf("%s has %d characters.\n", trimmed, count)
  }
}
